### PR TITLE
Eng/pr enable ondisk cas option

### PR DESCRIPTION
**Pull request by cachemeifyoucan**

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

File: llvm/CMakeLists.txt [hunk 1]
```diff
@@ -1282,6 +1283,13 @@ if (LLVM_INCLUDE_BENCHMARKS)
   add_subdirectory(benchmarks)
 endif()
 
+if (LLVM_ENABLE_ONDISK_CAS)
+  # Disable OnDisk CAS on windows and non-Unix systems.
+  if(WIN32 AND NOT UNIX)
```
Comment (benlangmuir) [inline llvm/CMakeLists.txt]: The comment implies this should be WIN32 **OR** NOT UNIX

File: llvm/include/llvm/CAS/CASReference.h [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 #include "llvm/ADT/ArrayRef.h"
 #include "llvm/ADT/DenseMapInfo.h"
+#include "llvm/ADT/Hashing.h"
```
Comment (benlangmuir) [inline llvm/include/llvm/CAS/CASReference.h]: What's this about?

Comment (cachemeifyoucan) [reply llvm/include/llvm/CAS/CASReference.h]: I should probably put this somewhere else but this is an attempt to fix this MSVC error:
```
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\llvm-project\llvm\include\llvm/CAS/CASReference.h(48): error C3861: 'hash_value': identifier not found
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\llvm-project\llvm\include\llvm/CAS/CASReference.h(48): error C2672: 'llvm::cas::hash_value': no matching overloaded function found
```

I don't think it worked here. Let me know if you have any idea for what happened here.

Comment (cachemeifyoucan) [reply llvm/include/llvm/CAS/CASReference.h]: ok I found a fix for MSVC.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
Windows swift bot builds after this change. There are some other clean up would be good to have but I will leave it for later commit.

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**